lib/browser/api/web-contents.ts

```diff
@@ -390,6 +390,17 @@ WebContents.prototype.getPrinters = function () {
   }
 };
 
+WebContents.prototype.getPrintersAsync = async function () {
```

Comment:
I'm not sure how I feel about introducing entirely new API surface here - can we in-place deprecate it and allow for both sync and async usage until the deprecated version has finished going through a deprecation cycle?

Reply:
@codebytere updated the PR to in-place deprecate the original variation of the function by accepting a boolean which switches the behavior of the function to return a promise instead, PTAL

Reply:
@codebytere I don't think changing the return value based on the parameter is a good idea, it will break existing TypeScript code:

```typescript
const printers = webContents.getPrinters()
for (const p of printers) {
}
```

```
error TS2488: Type '[] | Promise<[]>' must have a '[Symbol.iterator]()' method that returns an iterator.
```

Also the deprecation cycle will be weird:

1. First we print warning for `getPrinters()` calls without parameter.
2. After all code has migrated to `getPrinters(true)`, we remove the parameter and make `getPrinters()` async by default.
3. We either urge users to remove the redundant parameter by printing warnings, or all apps will stick with `getPrinters(true)`.